units/sdlthread.inc

```diff
@@ -66,27 +72,55 @@ type
 {$DEFINE SDL_PASSED_BEGINTHREAD_ENDTHREAD}
 
 type
-  {$IFNDEF DELPHI16UP}
-    TThreadID = Cardinal;
+  { SDL2-for-Pascal: #todo : Needed? }
+  {$IFNDEF FPC}
+    {$IFNDEF DELPHI16UP}
+      TThreadID = Cardinal;
+    {$ENDIF}
   {$ENDIF}
 
-  TpfnSDL_CurrentBeginThread = function(SecurityAttributes: Pointer; StackSize: LongWord; ThreadFunc: TThreadFunc; Parameter: Pointer; CreationFlags: LongWord; var ThreadId: TThreadID): cint;
+  { SDL2-for-Pascal #todo : Explanation needed }
+  TpfnSDL_CurrentBeginThread = function(
+    SecurityAttributes: Pointer; StackSize: LongWord; ThreadFunc: TThreadFunc;
+    Parameter: Pointer {arg}; CreationFlags: LongWord; var ThreadId: TThreadID {threadID}): cuintptr_t; cdecl;
```

Comment:
1. The signature here uses Pascal's LongWord. Should we use `cuint` instead? In the SDL headers (and Microsoft documentation) these arguments are of type `unsigned`.
2. In the C headers, the last argument is a pointer. Should we replace the pass-by-reference?

Reply:
1. I agree and fixed this.
2. I'm not sure why `ThreadFunc` and `ThreadID` rely on Pascal system definitions (`TThreadFunc` from RTL's `threadh.inc` and `TThreadID` from RTL's `sysosh.inc`) There is no pointer type of TThreadID defined, which means we would need to introduce it by ourselves which would be kind of ugly to mix in a pointer type definition for a RTL type within the SDL unit. 
 
I wonder if we should decouple this altogether like:
```
type
  TpfnSDL_CurrentBeginThreadFunc = function(params: Pointer):cuint;

  TpfnSDL_CurrentBeginThread = function(
    SecurityAttributes: Pointer; StackSize: cuint; func: TpfnSDL_CurrentBeginThreadFunc;
    Parameter: Pointer {arg}; CreationFlags: cuint; ThreadId: pcuint {threadID}): cuintptr_t; cdecl;
```